api/src/paths/project/{projectId}/survey/{surveyId}/sample-site/{surveySampleSiteId}/index.ts

```diff
@@ -279,6 +281,12 @@ export function deleteSurveySampleSiteRecord(): RequestHandler {
     try {
       await connection.open();
 
+      const observationService = new ObservationService(connection);
+
+      if ((await observationService.getObservationBySampleSiteId(surveyId, surveySampleSiteId)).length > 0) {
+        throw new HTTP400('Cannot delete sample site that has observations');
```

Comment:
might rephrase "Cannot delete a sample site that is associated with an observation"